File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int waterContain = 0;
        while(left<right){
            int width = right-left;
            int length = min(height[left],height[right]);
            int CurrentWater = length*width;
            waterContain = max(waterContain,CurrentWater);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return waterContain;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int WaterContain = 0;
        while(left<right){
            int length = Math.min(height[right],height[left]);
            int width = right - left;
            WaterContain = Math.max(WaterContain,length*width);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return WaterContain;
    }
}